app/src/PageController.php

```diff
@@ -3,9 +3,45 @@
 namespace {
 
     use SilverStripe\CMS\Controllers\ContentController;
+    use SilverStripe\Control\Cookie;
+    use SilverStripe\Control\Director;
+    use SilverStripe\Control\Session;
+    use SilverStripe\Core\Environment;
+    use SilverStripe\View\ArrayData;
+    use SilverStripe\View\Requirements;
 
     class PageController extends ContentController
     {
 
+        /**
+         * Whether our modal should be showed on this request.
+         * @var bool
+         */
+        private $showModal = false;
+
+        protected function init()
+        {
+            parent::init();
+
+            // Inject our privacy modal assets into the page
+            $cookie = Cookie::get('privacy_modal_showed');
+            if (empty($cookie)) {
+                $this->showModal = true;
+                Requirements::customScript('var privacy_modal_hostname=' . json_encode(Director::host()));
+                Requirements::javascript('app/js/dialog.js', ['defer' => true]);
+                Requirements::css('app/css/dialog.css');
+            }
+        }
+
+        /**
+         * Content of the Modal window if our show modal is enabled.
+         * @return \SilverStripe\ORM\FieldType\DBHTMLText|void
+         */
+        public function ModalWindow()
```

Comment:
So you've kept the name generic because that goes into the bambusa theme, but the specific implementation of a "privacy modal" is an application concern, right? it's a bit confusing, but I'd agree with that reasoning.

Reply:
Yeah that was the idea.